addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,719 @@
+        FROM account_invoice ai
+        WHERE am.old_invoice_id = ai.id AND ai.state not in ('draft', 'cancel')
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move (message_main_attachment_id, access_token,
+        name, date, ref, narration, type, journal_id, company_id, currency_id,
+        partner_id, commercial_partner_id, amount_untaxed, amount_tax,
+        amount_total, amount_residual, amount_untaxed_signed,
+        amount_total_signed, amount_residual_signed, fiscal_position_id,
+        invoice_user_id, state, invoice_payment_state, invoice_date,
+        invoice_date_due, invoice_payment_ref, invoice_sent, invoice_origin,
+        invoice_payment_term_id, invoice_partner_bank_id, invoice_incoterm_id,
+        invoice_source_email, invoice_partner_display_name,
+        invoice_cash_rounding_id, old_invoice_id, create_uid, create_date,
+        write_uid, write_date)
+        SELECT message_main_attachment_id, access_token,
+        COALESCE(number, move_name, name, '/'), COALESCE(date, date_invoice,
+        write_date), reference, comment, type, journal_id, company_id,
+        currency_id, partner_id, commercial_partner_id, amount_untaxed,
+        amount_tax, amount_total, residual, amount_untaxed_signed,
+        amount_total_signed, residual_signed, fiscal_position_id, user_id,
+        state, 'not_paid', date_invoice, date_due, name, sent, origin,
+        payment_term_id, partner_bank_id, incoterm_id, source_email,
+        vendor_display_name, cash_rounding_id, id, create_uid, create_date,
+        write_uid, write_date
+        FROM account_invoice ai
+        WHERE ai.state in ('draft', 'cancel')
+        RETURNING old_invoice_id, id""",
+    )
+    ids = env.cr.fetchall()
+    if ids:
+        _move_model_in_data(env, ids, 'account.invoice', 'account.move')
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move_line (company_id, account_id,
+        sequence, name, quantity, price_unit, discount,
+        price_subtotal, price_total, currency_id, partner_id, product_uom_id,
+        product_id, analytic_account_id, display_type, is_rounding_line,
+        move_id, old_invoice_line_id, create_uid, create_date, write_uid,
+        write_date)
+        SELECT ail.company_id, ail.account_id, ail.sequence, ail.name,
+        ail.quantity, ail.price_unit, ail.discount, ail.price_subtotal,
+        ail.price_total, ail.currency_id, ail.partner_id, ail.uom_id,
+        ail.product_id, ail.account_analytic_id, ail.display_type,
+        ail.is_rounding_line, COALESCE(ai.move_id, am.id), ail.id,
+        ail.create_uid, ail.create_date, ail.write_uid, ail.write_date
+        FROM account_invoice_line ail
+        LEFT JOIN account_invoice ai ON ail.invoice_id = ai.id
+        LEFT JOIN account_move am ON am.old_invoice_id = ai.id
+        WHERE COALESCE(ai.move_id, am.id) IS NOT NULL""",
```

Comment:
@MiquelRForgeFlow after migration I'm seeing duplicates move lines in validated invoices.
Don't we need to filter on `ai.state in ('draft', 'cancel')` here too?

Reply:
Also, this seems to create move lines with NULL debit and credit.

Reply:
The problem is that we cannot merge old move lines with old invoice lines, AFAIK, because there is not direct correspondence. Thus, at this moment what I do is mark old moves lines "with exclude_from_invoice_tab = TRUE" and transform old invoice lines to new move lines with "exclude_from_invoice_tab = False". Not sure if correct or if I should do something more/else. I am open to suggestions.

Reply:
For non draft invoices (ie invoices that have a move_id) a possibility could be to set `exclude_from_invoice_tab=TRUE` on move lines that are 
* on the payable/receivable account (which we know from the account field the invoice)
* or have `tax_line_id IS NOT NULL`

And for draft invoices I've not tested yet. Maybe it would be safer to let Odoo compute the move lines?

Reply:
and what about the move lines tat don't have those conditions? what about invoice lines?

Reply:
Normally these other move lines should be identical to invoice lines. Except when invoice line grouping was applied... in which case I don't see a solution to preserve the invoice lines.

Reply:
ok, can you check again a new migration?

Reply:
I will :)

Reply:
I'm still getting duplicate move lines on invoices (visible in the account move tab) with NULL debit and credit. I see the difficulty with merging invoice lines and move lines, and I don't see an easy solution for that, but I think that will cause troubles.